pypesto/ensemble/covariance_analysis.py

```diff
@@ -0,0 +1,318 @@
+    return get_covariance_matrix_lowlevel(dataset=ens.x_vectors.transpose())
+
+
+def get_covariance_matrix_predictions(
+        ens: Union[Ensemble, EnsemblePrediction],
+        prediction_index: int = 0) -> np.ndarray:
+    """
+    Compute the covariance of ensemble predictions.
+
+    Parameters
+    ==========
+    ens:
+        Ensemble object containing a set of parameter vectors and a set of
+        predictions or EnsemblePrediction object containing only predictions
+
+    prediction_index:
+        index telling which prediction from the list should be analyzed
+
+    Returns
+    =======
+    covariance_matrix:
+        covariance matrix of ensemble predictions
+    """
+
+    # extract the an array of predictions from either an Ensemble object or an
+    # EnsemblePrediction object
+    dataset = get_prediction_dataset(ens, prediction_index)
+
+    # call lowlevel routine using the prediction ensemble
+    return get_covariance_matrix_lowlevel(dataset=dataset)
+
+
+def get_spectral_decomposition_parameters(
+        ens: Ensemble,
+        normalize: bool = False,
+        lower_absolute_cutoff: float = None,
+        lower_relative_cutoff: float = None,
+        inv_upper_absolute_cutoff: float = None,
+        inv_upper_relative_cutoff: float = None,
+        only_identifiable_directions: bool = False,
+        only_separable_directions: bool = False):
+    """
+    Compute the spectral docmposition of ensemble parameters.
+
+    Parameters
+    ==========
+    ens:
+        Ensemble object containing a set of parameter vectors
+
+    normalize:
+        flag indicating whether the returned Eigenvalues should be normalized
+        with respect to the largest Eigenvalue
+
+    lower_absolute_cutoff:
+        Consider only eigenvalues of the covariance matrix above this cutoff
+        (only applied when only_separable_directions is True)
+
+    lower_relative_cutoff:
+        Consider only eigenvalues of the covariance matrix above this cutoff,
+        when rescaled with the largest eigenvalue
+        (only applied when only_separable_directions is True)
+
+    inv_upper_absolute_cutoff:
+        Consider only low eigenvalues of the covariance matrix with inverses
+        above of this cutoff
+        (only applied when only_identifiable_directions is True)
+
+    inv_upper_relative_cutoff:
+        Consider only low eigenvalues of the covariance matrix when rescaled
+        with the largest eigenvalue with inverses above of this cutoff
+        (only applied when only_identifiable_directions is True)
+
+    only_identifiable_directions:
+        return only identifiable directions according to inv_upper_cutoff
+
+    only_separable_directions:
+        return only separable directions according to lower_cutoff
+
+    Returns
+    =======
+    eigen_values:
+        Eigenvalues of the covariance matrix
+
+    eigen_vectors:
+        eigenvectors of the covariance matrix
```

Comment:
```suggestion
    A tuple with the following values in the following order.

    eigenvalues:
        Eigenvalues of the covariance matrix.

    eigenvectors:
        Eigenvectors of the covariance matrix.
```

Reply:
Should change `eigen_value` -> `eigenvalue` and `eigen_vector` -> `eigenvector` everywhere